### ISSUE TITLE
Variable names within function

### ISSUE THREAD

**Issue by davek44** [SheffieldML/GPyOpt#30]
I'd like to be able to use GPyOpt to model accuracy as a function of hyper-parameters for a machine learning algorithm run as a separate program through subprocess. To do so, I need access to the variables names in addition to their suggested values so that I can provide them unambiguously to the program. I can't figure out a good way to do this because my function that takes the suggested variable settings, runs the program, and returns the result cannot seem to see any outside variables, even when specified as global. Do you have any advice for how I might be able to make this work? Thanks!

**Comment by javiergonzalezh** [comment]
Not sure I completely understand your problem. Do you have any example to illustrate what you are trying to do?

Javier

**Comment by davek44** [comment]
Ah sorry, let me be more explicit. Let's say I have some hyper-parameters for a deep net, that I specify as:
```
hypers = [
{'name': 'learning_rate', 'type':'continuous', 'domain':(0.001, 0.01)},
{'name':'filters1', 'type':'continuous', 'domain':(100,500)},
{'name':'dropout1', 'type':'continuous', 'domain':(0,0.5)}
...etc...
]
```

I provide that list to GPyOpt.methods.BayesianOptimization along with my method to run the function as:
`
task = GPyOpt.methods.BayesianOptimization(run_ml, hypers)`

Where run_ml is a method that executes my program. Let's say it's a deep net in Torch.
```
def run_ml(suggest):
    # write suggested parameters to params.txt in table format "name\tvalue"
    subprocess.call('th deep_net.lua params.txt result.txt', shell=True)
    result = float(open('result.txt').readline())
    return result
```
I'm having trouble with that commented line where I'd like to write the suggested parameters to a file that the Torch code can read in, train a model, and print the result. But I don't have access to the hyper-parameter names "learning_rate", "filters1", "dropout1", etc in the run_ml method. I can hard code it anew everytime that I want to run the script, but I'd like it to be more generalizable than that, so I only have to write out the parameters and range one time. Thanks for helping me think through this.

**Comment by javiergonzalezh** [comment]
Ahh OK, I understand now. You are right, the objective function doesn't see the names of the variables, it does the right thing based on the order of the inputs. The solution I would try for your problem is to implement a wrapper class for ``run_ml`` with ``hypers`` in the ``__init__``. This way, whenever to run the line in which you want to write the parameters table you have access to the names. 

Perhaps this is what you already had in mind, but I can't thing now in a more generalizable way of doing it.
Hope it helps and thanks for using the package!

**Comment by davek44** [comment]
That worked for me. Thanks!
